app/models/concerns/archivable.rb

```diff
@@ -0,0 +1,12 @@
+module Archivable
+  extend ActiveSupport::Concern
+
+  included do
+    scope :archived, -> { where('archived_at is not null') }
+    scope :published, -> { where(archived_at: nil) }
```

Comment:
might need to rethink this.. published might be defined by the host model. in which case it might clash to define published like this. Maybe this should be not_archived, when can then be chained in to published as defined by the host.